### PR TITLE
DBW: update smokehouse tests to do deep equals

### PR DESCRIPTION
**Pull request by ebidel**
R: @brendankenny all

### CODE REVIEW

File: lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html [hunk 1]
```diff
@@ -345,6 +346,7 @@
 }
 </script>
 
+<!-- PASS: not in header, does not block rendering -->
```
Comment (patrickhulce) [inline lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html]: oh yeah actually could you comment that it's there to polyfill promise and make sure we don't break :)

**Comment by patrickhulce** [comment]
not a huge fan of the nested length object compared to `assert`/`expect`/`should`, feels a little hackier to me, but I'm fine for now until we actually support that.

File: lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html (hunk 1 above)
Comment (ebidel) [inline lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html]: done

**Comment by brendankenny** [comment]
🚬
🔦🔍

**Comment by ebidel** [comment]
Trav, you're my boyyyyyyy.

**Pull request merged**